crates/css-rs-macro/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+//! css-rs-macro is a procedural macro that allows you to write your CSS next to your Rust views.
+//!
+//! github.com/chinedufn/percy/examples/css-in-rust
+
+#[feature(proc_macro)]
+#[macro_use]
+extern crate syn;
+#[macro_use]
+extern crate quote;
+#[macro_use]
+extern crate lazy_static;
+extern crate proc_macro;
+
+use syn::Expr;
+
+use proc_macro::TokenStream;
+use proc_macro::TokenTree;
+
+use std::env;
+use std::fs;
+use std::fs::File;
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::path::Path;
+use std::sync::Mutex;
+
+lazy_static! {
+    static ref CSS_COUNTER: Mutex<u32> = Mutex::new(0);
+}
+
+/// Parses the syntax for writing inline css. Every call to css! will have its class
+/// name incremented by one.
+///
+/// So your first css! call is class "._css_rs_0", then "._css_rs_1", etc.
+///
+/// To write your css to a file use:
+///
+/// ```ignore
+/// OUTPUT_CSS=/path/to/my/output.css cargo run my-app
+/// ```
+///
+/// # Examples
+///
+/// ```ignore
+/// #![feature(use_extern_macros)]
+/// #![feature(proc_macro_non_items)]
+///
+/// extern crate css_rs_macro;
+/// use css_rs_macro::css;
+///
+/// fn main () {
+///     let class1 = css! {
+///       "
+///       :host {
+///         background-color: red;
+///       }
+///
+///       :host > div {
+///         display: flex;
+///         align-items: center;
+///       }
+///       "
+///     };
+///
+///     let class2 = css! {r#"
+///         :host { display: flex; }
+///     "#};
+///
+///     assert_eq!(class1, "_css_rs_0".to_string());
```

Comment:
In the future we will probably want a way to prevent class name collision when you have two different CSS in Rust css files being served to the same page for whatever reason.. maybe use a hash of some crate metadata as a prefix or something.. 

But doesn't seem like a concern to worry about right now.